react/features/video-menu/components/native/VolumeSlider.js

```diff
@@ -0,0 +1,122 @@
+// @flow
+
+import Slider from '@react-native-community/slider';
+import React, { Component } from 'react';
+import { View } from 'react-native';
+import { withTheme } from 'react-native-paper';
+
+import { Icon, IconVolumeEmpty } from '../../../base/icons';
+import { VOLUME_SLIDER_SCALE } from '../../constants';
+
+import styles from './styles';
+
+/**
+ * The type of the React {@code Component} props of {@link VolumeSlider}.
+ */
+type Props = {
+
+    /**
+     * The value of the audio slider should display at when the component first
+     * mounts. Changes will be stored in state. The value should be a number
+     * between 0 and 1.
+     */
+    initialValue: number,
+
+    /**
+     * The callback to invoke when the audio slider value changes.
+     */
+    onChange: Function,
+
+    /**
+     * Theme used for styles.
+     */
+    theme: Object
+};
+
+/**
+ * The type of the React {@code Component} state of {@link VolumeSlider}.
+ */
+type State = {
+
+    /**
+     * The volume of the participant's audio element. The value will
+     * be represented by a slider.
+     */
+    volumeLevel: number
+};
+
+/**
+ * Component that renders the volume slider.
+ *
+ * @returns {React$Element<any>}
+ */
+class VolumeSlider extends Component<Props, State> {
```

Comment:
Any reason for creating a class component instead of a functional one?

Reply:
The only reason was that the components created for this feature are class based.

Reply:
Could you please use a functional component? The ideea is to try to use functional components wherever possible and hooks. So instead of inheritance we could reuse code through hooks.

Reply:
We need to discuss this. I'm not 100% sure that this is a good idea. There can be performance issues and re-rendering for those components, we were debugging the participants' pane the last few days ...

Reply:
Updated

Reply:
@damencho There should be no difference in performance between class components & functional components. If there are multiple re-renders this must be because of how the component was built. I guess we should use React memo & Pure component more often.

Reply:
Yeah, I guess it was a bad implementation of the selectors as there were some places returning new objects (new constructed arrays) which will trigger new re-renders ... I'm not even sure where that was at the moment, we will soon create the PR.  
But there is a difference in how the comparison is done by react in the selectors it is === while in mapstatet2props it is deep equal comparison, so there is a big difference in this which can lead to re-renders.

Reply:
An example I just saw https://github.com/jitsi/jitsi-meet/blob/2174368d23e115781118b23ed6f00ce95ab30beb/react/features/av-moderation/functions.js#L95
This is used in a selector, this will be rendered on every action

Reply:
And maybe I added that, by the way :) not sure